### DISCUSSION TITLE
Side effects of `--no-ip-validation`

### DISCUSSION THREAD

**Discussion by lenzls** [allinurl/goaccess#2428] [Q&A]
Hello,

I want to see if I manage without logging IP addresses on my server.

I'm planning to use [ipscrub](https://github.com/masonicboom/ipscrub), to log only hashes of IP addresses. To do that, I assume I have to use `--no-ip-validation` as indicated in https://github.com/allinurl/goaccess/issues/1354.

I wonder, though, what are the costs of `--no-ip-validation`. I guess that geolocation doesn't work any more (unless with a workaround like https://github.com/allinurl/goaccess/issues/1089).
But will features like "unique visitors" still work with only the hash instead of the IP address?
Are there any other draw-backs of using `--no-ip-validation`?

Kind regards,
Simon

**Comment by allinurl** [comment]
Great question! If using `--no-ip-validation`, you would still need a placeholder for the `%h` field. The trade-off is accuracy. Very likely you will get a smaller number of visitors. I'd say give it a shot using a smaller number of requests, with and without IPs. Let me know what you get.

**Reply by lenzls** [reply]
Thanks for your reply. Small Update:

For now, I decided against hashing of the IP. Reason is, that there is an open issue in ipscrub which I don't understand https://github.com/masonicboom/ipscrub/issues/11.

Instead, I went with masking the last few bits of the IP address. As those are still valid IPs, I don't have to use the `--no-ip-validation` flag.

Note, I was surprised that the `--anonymize-ip` flag seems to only affect the rendered report, and not the persisted data (`--persist`). Since I don't want to indefinitely save non-masked IPs, I'm now doing the masking already on nginx side when writing the access.logs. But that is another topic :slightly_smiling_face:
